crates/telio-nurse/src/aggregator.rs

```diff
@@ -0,0 +1,113 @@
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    time::Instant,
+};
+
+use parking_lot::RwLock;
+use telio_crypto::PublicKey;
+use telio_model::api_config::EndpointProvider;
+use telio_utils::telio_log_warn;
+
+// Possible connection types - relayed ones allow adding info about the other endpoint
+#[derive(Copy, Clone, Debug)]
+pub enum ConnectionType {
+    Relayed,
+    Local(Option<EndpointProvider>),
+    UPnP(Option<EndpointProvider>),
+    UDPHolePunch(Option<EndpointProvider>),
+}
+
+// Possible disconnection reasons
+#[derive(Copy, Clone, Debug)]
+pub enum DisconnectionReason {
+    Initial,
```

Comment:
Is `initial` meant to be used in that startup phase, where we do not have connection yet. If that is indeed the purpose, maybe it is worth skipping the report for it at all?